### PR TITLE
SSS relaxation under ice

### PR DESCRIPTION
**Pull request by Jipingnersc**
Add one option for sss relaxation under ice in hycom_opt.

### CODE REVIEW

**Comment by lbertino** [comment]
Grammar: replace "at ice cover" by "below sea ice" or "where sea ice"

File: bin/ice_climatology/extract_clim_iceh.sh [hunk 1]
```diff
@@ -98,11 +98,6 @@ if [ -s ${Fini} -a -s ${Ftmpmask} ]; then
    ncrename -h -O -v siconc,aice_raw -v sithick,hi_raw ${Ftmp}  
    ncrename -h -O -v so,sss_raw -v thetao,sst_raw ${Ftmp}
```
Comment (TillRasmussen) [inline bin/ice_climatology/extract_clim_iceh.sh:100]: Ì  assuume that this section is only for cleaning up

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F [hunk 2]
```diff
@@ -14,6 +14,7 @@
       subroutine thermf_oi(m,n)
       use mod_xc         ! HYCOM communication interface
       use mod_cb_arrays  ! HYCOM saved arrays
+      use mod_NERSCnml, only : sss_underice, nersc_init
```
Comment (TillRasmussen) [inline hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:17]: You dont need to include nersc_init as it is not called from therm.F

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_NERSCnml.F90 [hunk 3]
```diff
@@ -46,8 +48,9 @@ subroutine NERSC_init
     end do
     close(funi)
     if (mnproc.eq.1) then
-      write (lp,*)'NERSC HYCOM: Reading hycom_nml from: hycom_opt'
+      write (lp,*)'NERSC HYCOM: Reading hycom_nml from: ../hycom_opt'
```
Comment (TillRasmussen) [inline hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_NERSCnml.F90:51]: I assume that this means that hycom_opt is located in the directory above SCRATCH (fine by me).

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F [hunk 4]
```diff
@@ -69,17 +70,20 @@ subroutine thermf_oi(m,n)
               surflx(i,j) = (1.0-covice(i,j))*surflx(i,j) +
      &                                        flxice(i,j)   !cell average
               sstflx(i,j) = (1.0-covice(i,j))*sstflx(i,j)   !relax over ocean
+              if (sss_underice==.false.) then
```
Comment (TillRasmussen) [inline hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:73]: If I read this correct.
When sst_underice=.false.
 Then before the sss relaxation was reduced by the ice concentration in points without rivers.
Now It is reduced by the ice concentration in all points.
If true then sss relaxation is never reduced due to the ice cover

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F (hunk 2 above)
Comment (TillRasmussen) [inline hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:17]: nersc init is not needed here.

**Review by TillRasmussen — COMMENTED** [review]
I added a few comments.

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_NERSCnml.F90 (hunk 3 above)
Comment (Jipingnersc) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_NERSCnml.F90:51]: I try to keep the same location as used by hycom_fabm.

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F (hunk 2 above)
Comment (Jipingnersc) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:17]: Thanks for this comment. It can be deleted.

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F (hunk 4 above)
Comment (Jipingnersc) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:73]: sst_underice has not been introduced because this model switches off the sst relaxation.

Comment (Jipingnersc) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:73]: So I understand you would like to comment sss_underice=.false. which meanings the sss relaxation was off (or reduced) due to ice cover.

Comment (lbertino) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:73]: Let me rephrase: I think that SSS relaxation should be off under ice cover.

Comment (TillRasmussen) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:73]: > sst_underice has not been introduced because this model switches off the sst relaxation.
Sorry typo. sss_underice...
To be more precise. This reduces the relaxation by the percentage of ice cover. For this purpose it works. An alternative could be to say if icecon>0.15 (example) ssrlx=0.

Comment (lbertino) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:73]: Personally I would prefer icecon>0.15 to the ice fraction in order to be more restrictive where the SSS climatology is bad. We may also have bad surprises when the model is run at lower resolution for longer periods.

Comment (Jipingnersc) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/thermf.F:73]: It will be a good constraint from my view also.
